cmd/snap/cmd_info.go

```diff
@@ -119,52 +121,22 @@ func maybePrintBase(w io.Writer, base string, verbose bool) {
 	}
 }
 
-func tryDirect(w io.Writer, path string, verbose bool, termWidth int) bool {
-	path = norm(path)
-
+func loadDirect(path string) (*client.Snap, error) {
 	snapf, err := snap.Open(path)
 	if err != nil {
-		return false
-	}
-
-	var sha3_384 string
-	if verbose && !osutil.IsDirectory(path) {
-		var err error
-		sha3_384, _, err = asserts.SnapFileSHA3_384(path)
-		if err != nil {
-			return false
-		}
+		return nil, err
 	}
-
 	info, err := snap.ReadInfoFromSnapFile(snapf, nil)
 	if err != nil {
-		return false
-	}
-	fmt.Fprintf(w, "path:\t%q\n", path)
-	fmt.Fprintf(w, "name:\t%s\n", info.InstanceName())
-	printSummary(w, info.Summary(), termWidth)
-
-	var notes *Notes
-	if verbose {
-		fmt.Fprintln(w, "notes:\t")
-		fmt.Fprintf(w, "  confinement:\t%s\n", info.Confinement)
-		if info.Broken == "" {
-			fmt.Fprintln(w, "  broken:\tfalse")
-		} else {
-			fmt.Fprintf(w, "  broken:\ttrue (%s)\n", info.Broken)
-		}
-
-	} else {
-		notes = NotesFromInfo(info)
+		return nil, err
 	}
-	fmt.Fprintf(w, "version:\t%s %s\n", info.Version, notes)
-	maybePrintType(w, string(info.Type))
-	maybePrintBase(w, info.Base, verbose)
-	if sha3_384 != "" {
-		fmt.Fprintf(w, "sha3-384:\t%s\n", sha3_384)
+
+	direct, err := cmd.ClientSnapFromSnapInfo(info)
```

Comment:
(not a concern of this PR) - feels a bit strange that `ClientSnapFromSnapInfo()` is under `cmd` …